toolchain/parser/testdata/pointer/fail_pointer_type_in_expression.carbon

```diff
@@ -0,0 +1,75 @@
+// CHECK:STDOUT:           {kind: 'Literal', text: '3'},
+// CHECK:STDOUT:           {kind: 'Literal', text: 'i32'},
+// CHECK:STDOUT:         {kind: 'InfixOperator', text: '*', subtree_size: 3},
+// CHECK:STDOUT:       {kind: 'PostfixOperator', text: '*', has_error: yes, subtree_size: 4},
+// CHECK:STDOUT:       {kind: 'Literal', text: '4'},
+// CHECK:STDOUT:     {kind: 'InfixOperator', text: '*', has_error: yes, subtree_size: 6},
+// CHECK:STDOUT:   {kind: 'ReturnStatement', text: ';', subtree_size: 8},
+// CHECK:STDOUT: {kind: 'FunctionDefinition', text: '}', subtree_size: 16},
+// CHECK:STDOUT:     {kind: 'FunctionIntroducer', text: 'fn'},
+// CHECK:STDOUT:     {kind: 'Name', text: 'G'},
+// CHECK:STDOUT:       {kind: 'ParameterListStart', text: '('},
+// CHECK:STDOUT:     {kind: 'ParameterList', text: ')', subtree_size: 2},
+// CHECK:STDOUT:       {kind: 'Literal', text: 'i32'},
+// CHECK:STDOUT:     {kind: 'ReturnType', text: '->', subtree_size: 2},
+// CHECK:STDOUT:   {kind: 'FunctionDefinitionStart', text: '{', subtree_size: 7},
+// CHECK:STDOUT:     {kind: 'ReturnStatementStart', text: 'return'},
+// CHECK:STDOUT:         {kind: 'Literal', text: 'i32'},
+// CHECK:STDOUT:       {kind: 'PostfixOperator', text: '*', subtree_size: 2},
+// CHECK:STDOUT:       {kind: 'Literal', text: '4'},
+// CHECK:STDOUT:     {kind: 'InfixOperator', text: '+', has_error: yes, subtree_size: 4},
+// CHECK:STDOUT:   {kind: 'ReturnStatement', text: ';', subtree_size: 6},
+// CHECK:STDOUT: {kind: 'FunctionDefinition', text: '}', subtree_size: 14},
+// CHECK:STDOUT:     {kind: 'FunctionIntroducer', text: 'fn'},
+// CHECK:STDOUT:     {kind: 'Name', text: 'H'},
+// CHECK:STDOUT:       {kind: 'ParameterListStart', text: '('},
+// CHECK:STDOUT:     {kind: 'ParameterList', text: ')', subtree_size: 2},
+// CHECK:STDOUT:       {kind: 'Literal', text: 'i32'},
+// CHECK:STDOUT:     {kind: 'ReturnType', text: '->', subtree_size: 2},
+// CHECK:STDOUT:   {kind: 'FunctionDefinitionStart', text: '{', subtree_size: 7},
+// CHECK:STDOUT:     {kind: 'ReturnStatementStart', text: 'return'},
+// CHECK:STDOUT:         {kind: 'Literal', text: 'i32'},
+// CHECK:STDOUT:       {kind: 'PrefixOperator', text: '*', subtree_size: 2},
+// CHECK:STDOUT:     {kind: 'PostfixOperator', text: '*', has_error: yes, subtree_size: 3},
+// CHECK:STDOUT:   {kind: 'ReturnStatement', text: ';', subtree_size: 5},
+// CHECK:STDOUT: {kind: 'FunctionDefinition', text: '}', subtree_size: 13},
+// CHECK:STDOUT: {kind: 'FileEnd', text: ''},
+// CHECK:STDOUT: ]
+
+fn F() -> i32 {
+  // CHECK:STDERR: fail_pointer_type_in_expression.carbon:[[@LINE+6]]:17: Parentheses are required to disambiguate operator precedence.
+  // CHECK:STDERR:   return 3 * i32* * 4;
+  // CHECK:STDERR:                 ^
+  // CHECK:STDERR: fail_pointer_type_in_expression.carbon:[[@LINE+3]]:19: Parentheses are required to disambiguate operator precedence.
+  // CHECK:STDERR:   return 3 * i32* * 4;
+  // CHECK:STDERR:                   ^
```

Comment:
This seems a little undesirable to repeat. Maybe we should have a TODO to better propagate errors in expression precedence?

Reply:
Done.